src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const changedState = { ...state };
+  const stateVersions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(changedState, action.extraData);
+        stateVersions.push({ ...changedState });
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete changedState[property];
+        }
+        stateVersions.push({ ...changedState });
+        break;
+
+      case 'clear':
+        const keys = Object.keys(changedState);
+
+        for (const key of keys) {
+          delete changedState[key];
+        }
+        stateVersions.push({ ...changedState });
+        break;
```

Comment:
ch#4 [DONT REPEAT YOURSELF] you used "push" x 3 times, try differen option

Reply:
fixed.